### ISSUE TITLE
tflearn questions.

### ISSUE THREAD

**Issue by psnegi** [tflearn/tflearn#225]
I have  couple of questions about tflearn

1) How to use own python function for data preprocessing and data augmentation.
2) In conv_2d is it possible to add a batch_normalization key, value argument so that 
  batch normalization happen before applying activation function.

3) In regression  function there is an ambiguity using parameters   optimizer='optimizer name', and learning_rate=0.001 when using optimizer tensor instead of string name for optimizer. 
Which learning rate will be effective? is it the one used for building optimizer or one used in regression
function as default learning rate parameter.
Thanks,

**Comment by aymericdamien** [comment]
1) To use your own function, you have to directly modify tflearn source code and add your methods
2) There is no batch_norm argument, so you need to do `tflearn.relu(tflearn.batch_normalization(conv_2d(...)))`
3) Learning rate parameter is only used if you are giving optimizer parameters as a string. If you are using an optimizer tensor, it will use the tensor's learning rate.
